packages/core/src/checks/search-is-array.js

```diff
@@ -5,19 +5,53 @@ const { simpleTruncate } = require('../tools/data');
 
 const isSearch = require('./is-search');
 
+const hasCanPaginate = (searchKey, compiledApp) => {
+  const canPaginate = _.get(compiledApp, [
+    'searches',
+    searchKey,
+    'operation',
+    'canPaginate',
+  ]);
+  return canPaginate;
+};
+
 /*
-  Searches should always return an array of objects.
+  Searches should return an array of objects,
+  or an object like { results: [...], paging_token: '...' } when canPaginate is true.
 */
-const searchIsArray = {
-  name: 'triggerIsArray',
+const searchIsArrayOrObject = {
+  name: 'searchIsArrayOrObject',
   shouldRun: isSearch,
-  run: (method, results) => {
+  run: (method, results, compiledApp) => {
+    console.log('searchIsArray', method, results, 'compiledApp', compiledApp);
+
+    const searchKey = method.split('.', 2)[1];
+    const truncatedResults = simpleTruncate(JSON.stringify(results), 50);
+
+    if (hasCanPaginate(searchKey, compiledApp)) {
+      // must be an object
+      if (!_.isObject(results)) {
+        return [
+          `Paginating search results must be an object, got: ${typeof results}, (${truncatedResults})`,
+        ];
+      }
+      // must have results and paging_token
+      if (!_.has(results, 'results') || !_.has(results, 'paging_token')) {
```

Comment:
Do we also need to check that `paging_token` is a string or `null` (or not/undefined?) and that `results` is an `Array` (of objects)?

Reply:
I'm setting `results = results.results` so that the existing check for an array can take care of that 👍 Re: `paging_token`, yes, I've added a check for `paging_token` to be a string.